spec.bs

```diff
@@ -5377,15 +5438,82 @@ result of [=evaluating a bidding script=], or an [=additional bid=] provided by
   : <dfn>component seller</dfn>
   :: Null or an [=origin=]. Seller in component auction which the [=generated bid/interest group=]
     is participating. Only set for component auctions, null otherwise.
+  : <dfn>target number of ad components</dfn>
+  :: Null or {{unsigned long}}, initially null. Set if the bidder requested
+    [=generated bid/ad component descriptors=] they provided to be reduced to the given target,
+    taking in account k-anonymity.
+  : <dfn>number of mandatory ad components</dfn>
+  :: {{unsigned long}}. Only relevant if [=generated bid/target number of ad components=] is set.
+     Requires the first that many given ad components to be included.
 </dl>
 
+<div algorithm>
+
+To <dfn>apply any component ads target to a bid</dfn>, given a [=generated bid=] |generatedBid|:
+1. If |generatedBid|'s [=generated bid/target number of ad components=] is null, return.
+1. [=Assert=] that |generatedBid|'s [=generated bid/ad component descriptors=] is not null.
+1. While |generatedBid|'s [=generated bid/ad component descriptors=]'s [=list/size=] &gt;
+    [=generated bid/target number of ad components=]:
+    1. [=list/Remove=] the last value from |generatedBid|'s [=generated bid/ad component descriptors=].
+
+</div>
+
+<div algorithm>
+
+To <dfn>try to reach component ads target considering k-anonymity</dfn>, given a [=generated bid=] |generatedBid|:
+1. If |generatedBid|'s [=generated bid/target number of ad components=] is null, return false.
+1. Let |selectedComponents| be a new [=list=] of [=ad descriptors=].
+1. [=set/For each=] |i| of [=list/get the indices=] of |generatedBid|:
+  1. Let |candidateComponent| be |generatedBid|'s [=generated bid/ad component descriptors=][|i|].
+  1. If [=query component ad k-anonymity count=] given |candidateComponent|'s [=interest group ad/render url=] returns true:
+
+      Issue: TODO: change to query k-anonymity cache instead, once

```

Comment:
Can you file an issue for this? For anything substantial, I prefer issues to be filed to not lose track of things.

Reply:
https://github.com/WICG/turtledove/issues/1150.
@morlovich You can add a link to this issue. See [this example](https://github.com/WICG/turtledove/blob/main/spec.bs#L5350).

Reply:
Thanks Qingxin; done.